File: List Problems/two_sum.py
#hint if is[i] = x then seaarch for element that is target - x
def twoSum(nums, target): #TC is O(N2)
    
    for i in range(len(nums)):
        
        compliment = target - nums[i]
        if nums[i+1] == nums[i] and nums[i] + nums[i+1] == target:
            return [i+1, i+2]
        if compliment in nums and i != nums[i]:
            return [i+1, nums.index(compliment)+1]
        

    
    

nums = [1,2,3,4,4,9,56,90]


print(twoSum(nums, 8))

File: LEETCODE/Arrays/two_sum.cpp
#include<iostream>
#include<unordered_map>
#include<vector>
using namespace std;

int main(){
    vector<int> nums = {2, 3, 7, 8, 23, 12, 67, 54, 23};
    int target = 61;

    unordered_map<int, int> mp;
    for(int i = 0 ; i < nums.size(); i++){
        if(mp.find(target - nums[i]) == mp.end()){
            mp[nums[i]] = i;
        }
        else{
            cout << "(" << mp[target - nums[i]] << " , " << i << ")";
        }
    }

    return 0;

}